pagerduty/data_source_pagerduty_extension_schema.go

```diff
@@ -33,7 +33,7 @@ func dataSourcePagerDutyExtensionSchemaRead(d *schema.ResourceData, meta interfa
 
 	searchName := d.Get("name").(string)
 
-	resp, _, err := client.ExtensionSchemas.List()
+	resp, _, err := client.ExtensionSchemas.List(&pagerduty.ListExtensionSchemasOptions{Query: searchName})
```

Comment:
Does the regex match on label functionality no longer get used if this query is explicit? If so, should probably be removed.

Reply:
I totally agree, we should actually be able to drop the regex matching altogether.

One idea would be to replace the current regex matching with something like this:

```go
    var found *pagerduty.ExtensionSchema

    for _, es := range resp.ExtensionSchemas {
        if strings.EqualFold(es.Label, searchName) {
            found = es
        }
    }

    if found == nil {
        return fmt.Errorf("Unable to locate any extension schema with the name: %s", searchName)
    }
```

to _still_ allow case-insensitive matching (such as **slack**/**Slack**).

What do you think?

Reply:
Shouldn't this datasource ensure that there is exactly one match?

That, or return a list, with the benefit of returning an empty list instead of erroring when no match is found.

Reply:
Typically the pattern we use is have a separate data source (with a plural name) when you want to return a list, and have the singular data source error if multiple matches are found.

That being said, I think EqualFold would be great (didn't even know that function existed until now!) as long as Pager Duty is also case insensitive when we use the Query? If thats the case though, I'd say go ahead and make that change.

Reply:
That sounds good to me :+1:

PagerDuty does indeed seem to be case insensitive, 
so I'll go ahead and submit the change with `EqualFold`.

Thank you both so much for helping me out on this one :pray: